receiver/awscontainerinsightreceiver/internal/k8sapiserver/prometheus_scraper.go

```diff
@@ -19,74 +19,150 @@ import (
+)
+
 type PrometheusScraper struct {
-	ctx                      context.Context
-	settings                 component.TelemetrySettings
-	host                     component.Host
-	clusterNameProvider      clusterNameProvider
-	simplePrometheusReceiver receiver.Metrics
-	leaderElection           *LeaderElection
-	running                  bool
+	ctx                 context.Context
+	settings            component.TelemetrySettings
+	host                component.Host
+	clusterNameProvider clusterNameProvider
+	prometheusReceiver  receiver.Metrics
+	leaderElection      *LeaderElection
+	running             bool
+}
+
+type PrometheusScraperOpts struct {
+	Ctx                 context.Context
+	TelemetrySettings   component.TelemetrySettings
+	Endpoint            string
+	Consumer            consumer.Metrics
+	Host                component.Host
+	ClusterNameProvider clusterNameProvider
+	LeaderElection      *LeaderElection
+	BearerToken         string
 }
 
-func NewPrometheusScraper(ctx context.Context, telemetrySettings component.TelemetrySettings, endpoint string, nextConsumer consumer.Metrics, host component.Host, clusterNameProvider clusterNameProvider, leaderElection *LeaderElection) (*PrometheusScraper, error) {
-	if leaderElection == nil {
-		return nil, errors.New("leader election cannot be null")
+func NewPrometheusScraper(opts PrometheusScraperOpts) (*PrometheusScraper, error) {
+	if opts.Consumer == nil {
+		return nil, errors.New("consumer cannot be nil")
+	}
+	if opts.Host == nil {
+		return nil, errors.New("host cannot be nil")
+	}
+	if opts.LeaderElection == nil {
+		return nil, errors.New("leader election cannot be nil")
+	}
+	if opts.ClusterNameProvider == nil {
+		return nil, errors.New("cluster name provider cannot be nil")
 	}
 
-	spConfig := simpleprometheusreceiver.Config{
-		HTTPClientSettings: confighttp.HTTPClientSettings{
-			Endpoint: endpoint,
-			TLSSetting: configtls.TLSClientSetting{
-				TLSSetting: configtls.TLSSetting{
-					CAFile: caFile,
-				},
-				Insecure:           false,
+	controlPlaneMetricsAllowRegex := ""
+	for _, item := range controlPlaneMetricAllowList {
+		controlPlaneMetricsAllowRegex += item + "|"
+	}
+	controlPlaneMetricsAllowRegex = strings.TrimSuffix(controlPlaneMetricsAllowRegex, "|")
+
+	scrapeConfig := &config.ScrapeConfig{
+		HTTPClientConfig: configutil.HTTPClientConfig{
+			TLSConfig: configutil.TLSConfig{
+				CAFile:             caFile,
 				InsecureSkipVerify: false,
 			},
 		},
-		MetricsPath:        "/metrics",
-		CollectionInterval: collectionInterval,
-		UseServiceAccount:  true,
+		ScrapeInterval:  model.Duration(collectionInterval),
+		ScrapeTimeout:   model.Duration(collectionInterval),
+		JobName:         fmt.Sprintf("%s/%s", "containerInsightsKubeAPIServerScraper", opts.Endpoint),
+		HonorTimestamps: true,
+		Scheme:          "https",
+		MetricsPath:     "/metrics",
+		ServiceDiscoveryConfigs: discovery.Configs{
+			&discovery.StaticConfig{
+				{
+					Targets: []model.LabelSet{
```

Comment:
Dont need to change now, but if we see this config growing big in the future, we could actually keep this in a yaml and embed it, deserialize that into a config and set some of the properties that exist here in code as overrides.

Reply:
Could also do placeholders in the yaml and once we read the embed string, replace placeholders with actual values. For ex, have a placeholder like `<ClusterName>` which gets replaced with the actual cluster name in code.

Reply:
That makes sense.  I think once we get to two scrape sources then we should refactor